src/modules/dom.js

```diff
@@ -0,0 +1,306 @@
+
+    comments.innerHTML = '';
+    div1.innerHTML = '';
+    div2.innerHTML = '';
+    const createData = {
+      type: ['p', null, null],
+      status: ['p', null, null],
+      lang: ['p', null, null],
+      genre: ['p', null, null],
+      premiere: ['p', null, null],
+      end: ['p', null, null],
+    };
+
+    const elem = this.batchCreateElements(createData);
+    loader.url = `https://api.tvmaze.com/search/shows?q=${name}`;
+    const data = await loader.getData();
+    const appendData = [
+      { child: elem.type, parent: div1 },
+      { child: elem.status, parent: div1 },
+      { child: elem.lang, parent: div1 },
+      { child: elem.genre, parent: div2 },
+      { child: elem.premiere, parent: div2 },
+      { child: elem.end, parent: div2 },
+    ];
+    this.batchAppendElements(appendData, false);
+
+    const showName = data[0].show.name;
+
+    nameShow.innerHTML = showName;
+    imageShow.src = data[0].show.image.medium;
+    elem.type.innerHTML = `Type: ${data[0].show.type}`;
+    elem.status.innerHTML = `Status: ${data[0].show.status}`;
+    elem.lang.innerHTML = `Language: ${data[0].show.language}`;
+    elem.genre.innerHTML = `Genres: ${data[0].show.genres.join(', ')}`;
+    elem.premiere.innerHTML = `Premiered at: ${data[0].show.premiered}`;
+    elem.end.innerHTML = `Ended at: ${data[0].show.ended}`;
+
+    const noComments = await this.commentCounter(showName);
+    commentHeader.innerHTML = `Comments(${noComments})`;
+    this.submitComment(showName);
+  };
+
+  likeItem = (likebutton, id) => {
+    likebutton.addEventListener('click', async (event) => {
+      event.preventDefault();
+      loader.url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/DXeDvMiVvdYqzelJx7jP/likes';
+      const itemId = {
+        item_id: id,
+      };
+      const likes = await loader.setData(itemId);
+      if (likes === 201) {
+        const lk = likebutton.parentNode.childNodes[1].innerHTML.split(' ')[0];
+        likebutton.parentNode.childNodes[1].innerHTML = `${
+          Number(lk) + 1
+        } likes`;
+      }
+    });
+  };
+
+  loadLikes = async () => {
+    loader.url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/DXeDvMiVvdYqzelJx7jP/likes';
+    const likes = await loader.getData();
+    return likes;
+  };
+
+  addComment = (commentData) => {
+    loader.url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/DXeDvMiVvdYqzelJx7jP/comments';
+    const comment = loader.setData(commentData);
+    return comment;
+  };
+
+  getComment = async (itemId) => {
+    loader.url = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/DXeDvMiVvdYqzelJx7jP/comments?item_id=${itemId}`;
+    let comment = await loader.getData();
+    if (comment.error && comment.error.status > 210) {
+      comment = null;
+    }
+    this.createComments(comment);
+    if (comment === null) {
+      return 0;
+    }
+    return comment.length;
```

Comment:
According to the requirements, the counter is supposed to be implemented as a separate module and the absolute length of the API should not be used as the counter, instead you are expected to look for specific elements in the DOM (you can utilize querySelectorAll for this) in order to achieve this. Kindly implement this as described in order to stick with the project requirements.